module/os/linux/spl/spl-taskq.c

```diff
@@ -491,6 +491,13 @@ taskq_member(taskq_t *tq, kthread_t *t)
 }
 EXPORT_SYMBOL(taskq_member);
 
+taskq_t *
+taskq_of_curthread(void)
+{
+	return (tsd_get(taskq_tsd));
+}
+EXPORT_SYMBOL(taskq_of_curthread);
```

Comment:
Makes good sense.  My only minor concern is other platforms will need to add this new taskq interface.  That looks like it will be trivial for illumos but may be somewhat problematic for FreeBSD (though I may be missing something).  Let's find out how difficult adding this would be.

Reply:
That satisfies my concern.  I'm good with this.